Sources/Clock2.swift

```diff
@@ -122,8 +125,8 @@ public enum Clock2 {
         NTPClient().query(pool: pool, numberOfSamples: samples) { offset, done, total in
             assert(Thread.isMainThread)
             if let offset = offset {
-                storage.stableTime = TimeFreeze(offset: offset)
                 protection.lock()
+                storage.stableTime = TimeFreeze(offset: offset)
```

Comment:
It seems better to move out initialization to out of critical section to minimize critical sections.
```suggestion
                let newFreeze = TimeFreeze(offset: offset)
                protection.lock()
                storage.stableTime = newFreeze
                latestOffset = offset
                protection.unlock()
```

Also, at this point, it's impossible to find out thread-safety information only from code in `systemUptime` function.
Note (or link) thread-safety characteristics on each line of call to sys-calls. So future maintainers can find out them easily.

Reply:
Thanks! I have updated the code!